auth/src/main/java/com/firebase/ui/auth/data/client/CountryListLoadTask.java

```diff
@@ -38,267 +42,52 @@
-        countryInfoList.add(new CountryInfo(new Locale("", "SJ"), 47));
-        countryInfoList.add(new CountryInfo(new Locale("", "SZ"), 268));
-        countryInfoList.add(new CountryInfo(new Locale("", "SE"), 46));
-        countryInfoList.add(new CountryInfo(new Locale("", "CH"), 41));
-        countryInfoList.add(new CountryInfo(new Locale("", "SY"), 963));
-        countryInfoList.add(new CountryInfo(new Locale("", "TW"), 886));
-        countryInfoList.add(new CountryInfo(new Locale("", "TJ"), 992));
-        countryInfoList.add(new CountryInfo(new Locale("", "TZ"), 255));
-        countryInfoList.add(new CountryInfo(new Locale("", "TH"), 66));
-        countryInfoList.add(new CountryInfo(new Locale("", "TG"), 228));
-        countryInfoList.add(new CountryInfo(new Locale("", "TK"), 690));
-        countryInfoList.add(new CountryInfo(new Locale("", "TO"), 676));
-        countryInfoList.add(new CountryInfo(new Locale("", "TT"), 1));
-        countryInfoList.add(new CountryInfo(new Locale("", "TN"), 216));
-        countryInfoList.add(new CountryInfo(new Locale("", "TR"), 90));
-        countryInfoList.add(new CountryInfo(new Locale("", "TM"), 993));
-        countryInfoList.add(new CountryInfo(new Locale("", "TC"), 1));
-        countryInfoList.add(new CountryInfo(new Locale("", "TV"), 688));
-        countryInfoList.add(new CountryInfo(new Locale("", "VI"), 1));
-        countryInfoList.add(new CountryInfo(new Locale("", "UG"), 256));
-        countryInfoList.add(new CountryInfo(new Locale("", "UA"), 380));
-        countryInfoList.add(new CountryInfo(new Locale("", "AE"), 971));
-        countryInfoList.add(new CountryInfo(new Locale("", "GB"), 44));
-        countryInfoList.add(new CountryInfo(new Locale("", "US"), 1));
-        countryInfoList.add(new CountryInfo(new Locale("", "UY"), 598));
-        countryInfoList.add(new CountryInfo(new Locale("", "UZ"), 998));
-        countryInfoList.add(new CountryInfo(new Locale("", "VU"), 678));
-        countryInfoList.add(new CountryInfo(new Locale("", "VA"), 379));
-        countryInfoList.add(new CountryInfo(new Locale("", "VE"), 58));
-        countryInfoList.add(new CountryInfo(new Locale("", "VN"), 84));
-        countryInfoList.add(new CountryInfo(new Locale("", "WF"), 681));
-        countryInfoList.add(new CountryInfo(new Locale("", "EH"), 212));
-        countryInfoList.add(new CountryInfo(new Locale("", "YE"), 967));
-        countryInfoList.add(new CountryInfo(new Locale("", "ZM"), 260));
-        countryInfoList.add(new CountryInfo(new Locale("", "ZW"), 263));
+        List<CountryInfo> countryInfoList = getAvailableCountryCodes();
         Collections.sort(countryInfoList);
         return countryInfoList;
     }
 
+    public List<CountryInfo> getAvailableCountryCodes() {
+        Map<String, Integer> countryInfoMap = PhoneNumberUtils.getImmutableCountryCodeMap();
+        if (whitelistedCountryCodes == null && blacklistedCountryCodes == null) {
```

Comment:
I think if you want this to be the default you should do this in the constructor and then let every other method in the class assume that the whitelist and blacklist are setup properly.

Reply:
Might also want to throw an `IllegalStateException` in the constructor if they're both non-null.

Reply:
I'm not sure if I agree with adding IllegalStateException. I moved that bit to the constructor though.